src/ui/layout/pane.vue

```diff
@@ -70,20 +71,55 @@ export default {
         this.type = this.$parent.type;
         this.styleProp = (this.type === 'horizontal') ? 'width' : 'height';
     },
+    async mounted() {
+        await this.$nextTick();
+        // Hide tree and/or inspector pane if specified in URL
+        this.handleHideUrl();
+    },
     methods: {
+
         toggleCollapse: function () {
             this.collapsed = !this.collapsed;
             if (this.collapsed) {
-                // Pane is expanded and is being collapsed
-                this.currentSize = (this.dragCollapse === true) ? this.initial : this.$el.style[this.styleProp];
-                this.$el.style[this.styleProp] = '';
+                this.handleCollapse();
             } else {
-                // Pane is collapsed and is being expanded
-                this.$el.style[this.styleProp] = this.currentSize;
-                delete this.currentSize;
-                delete this.dragCollapse;
+                this.handleExpand();
+            }
+        },
+        handleHideUrl: function () {
+            const HIDE_TREE_PARAM = 'hideTree';
+            const HIDE_INSPECTOR_PARAM = 'hideInspector';
+            const TREE_PANE_CLASS_NAME = 'l-shell__pane-tree';
+            const INSPECTOR_PANE_CLASS_NAME = 'l-shell__pane-inspector';
+            let hideTree = this.openmct.router.getSearchParam(HIDE_TREE_PARAM);
+            let hideInspector = this.openmct.router.getSearchParam(HIDE_INSPECTOR_PARAM);
+            if (hideTree) {
+                this.collapsePane(TREE_PANE_CLASS_NAME, HIDE_TREE_PARAM);
+            }
+
+            if (hideInspector) {
+                this.collapsePane(INSPECTOR_PANE_CLASS_NAME, HIDE_INSPECTOR_PARAM);
             }
         },
+        collapsePane: function (PaneClass, param) {
+            for (let key of this.$el.classList) {
+                if (key === PaneClass && this.collapsable) {
+                    this.collapsed = true;
+                    this.handleCollapse();
+                    this.openmct.router.deleteSearchParam(param);
+                    break;
+                }
+            }
+        },
+        handleCollapse: function () {
```

Comment:
Can we use the existing `toggleCollapse` method here instead?

Reply:
Hi @shefalijoshi  the toggleCollapse handles both collapse and expand. It calls either handleCollapse or handleExpand.